From 618fd7e2de9571c8bcab6c95c671662d9e45e6a3 Mon Sep 17 00:00:00 2001
From: Richard Bradley <richard.bradley@joyent.com>
Date: Wed, 12 Apr 2017 09:13:56 +0100
Subject: [PATCH] MANTA-2587 manta-net.sh Needs to Support Adding tags to
 Aggregations MANTA-3014 Only distribute manta-nic and xdc-routes SMF services
 if required

---
 docs/operator-guide/index.md | 66 ++++++++++++++++++++++++++++--------
 1 file changed, 52 insertions(+), 14 deletions(-)

diff --git a/docs/operator-guide/index.md b/docs/operator-guide/index.md
index 37270fd..4e31dda 100644
--- a/docs/operator-guide/index.md
+++ b/docs/operator-guide/index.md
@@ -833,16 +833,16 @@ for details on how to check if services are healthy.
 The networking configuration file is a per-datacenter JSON file with several
 properties:
 
-| Property       | Kind                       | Description                                                                                                                                                                                   |
-| -------------- | -------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
-| `azs`          | array&nbsp;of&nbsp;strings | list of all availabililty zones (datacenters) participating in Manta in this region                                                                                                           |
-| `this_az`      | string                     | string (in `azs`) denoting this availability zone                                                                                                                                             |
-| `manta_nodes`  | array&nbsp;of&nbsp;strings | list of server uuid's for *all* servers participating in Manta in this AZ                                                                                                                     |
-| `marlin_nodes` | array&nbsp;of&nbsp;strings | list of server uuid's (subset of `manta_nodes`) that are storage nodes                                                                                                                        |
-| `admin`        | object                     | describes the "admin" network in this datacenter (see below)                                                                                                                                  |
-| `manta`        | object                     | describes the "manta" network in this datacenter (see below)                                                                                                                                  |
-| `marlin`       | object                     | describes the "marlin" network in this datacenter (see below)                                                                                                                                 |
-| `mac_mappings` | object                     | maps each server uuid from `manta_nodes` to an object mapping each network name ("admin", "manta", and "marlin") to the MAC address on that server over which that network should be created. |
+| Property       | Kind                       | Description                                                                                                                                                                          |
+| -------------- | -------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| `azs`          | array&nbsp;of&nbsp;strings | list of all availabililty zones (datacenters) participating in Manta in this region                                                                                                  |
+| `this_az`      | string                     | string (in `azs`) denoting this availability zone                                                                                                                                    |
+| `manta_nodes`  | array&nbsp;of&nbsp;strings | list of server uuid's for *all* servers participating in Manta in this AZ                                                                                                            |
+| `marlin_nodes` | array&nbsp;of&nbsp;strings | list of server uuid's (subset of `manta_nodes`) that are storage nodes                                                                                                               |
+| `admin`        | object                     | describes the "admin" network in this datacenter (see below)                                                                                                                         |
+| `manta`        | object                     | describes the "manta" network in this datacenter (see below)                                                                                                                         |
+| `marlin`       | object                     | describes the "marlin" network in this datacenter (see below)                                                                                                                        |
+| `nic_mappings` | object                     | maps each server in both `manta_nodes` and `marlin_nodes` to an object mapping each network name ("manta" and "marlin") to the network interface on the server that should be tagged |
 
 "admin", "manta", and "marlin" all describe these networks that are built into Manta:
 
@@ -865,6 +865,25 @@ Besides those two, each of these blocks has a property for the current
 availability zone that describes the "subnet", "gateway", "vlan_id", and
 "start" and "end" provisionable addresses.
 
+`nic_mappings` is a nested object structure defining the network interface to be
+tagged for each server defined in `manta_nodes` and `marlin_nodes`, and for each
+of Manta's built in networks. See below for an example of this section of the
+configuration.
+
+Note: Aggregations must already exist in NAPI, and updating NIC tags on
+aggregations will require a reboot of the server in question.
+
+The optional boolean `distribute_svcs` gives control to the operator over the
+boot-time networking detection that happens each time `manta-net.sh` is executed
+(which determines if the global zone SMF services should be distributed). For
+example, an operator has enabled boot-time networking in a datacenter _after_
+installing Manta, and subsequently would like to add some more storage nodes.
+For consistency, the operator can set `distribute_svcs` to `true` in order to
+force distribution of these global zone services.
+
+Note: For global zone network changes handled by boot-time networking to
+take effect a reboot of the node must be performed.
+
 For reference, here's an example multi-datacenter configuration with one service
 node (aac3c402-3047-11e3-b451-002590c57864) and one storage node
 (445aab6c-3048-11e3-9816-002590c3f3bc):
@@ -954,15 +973,34 @@ node (aac3c402-3047-11e3-b451-002590c57864) and one storage node
         }
       },
 
-      "mac_mappings": {
+      "nic_mappings": {
         "aac3c402-3047-11e3-b451-002590c57864": {
-          "manta": "90:e2:ba:4b:ec:d1"
+          "manta": {
+            "mac": "90:e2:ba:4b:ec:d1"
+          }
         },
         "445aab6c-3048-11e3-9816-002590c3f3bc": {
-          "manta": "90:e2:ba:4a:32:71",
-          "mantanat": "90:e2:ba:4a:32:71"
+          "manta": {
+            "mac": "90:e2:ba:4a:32:71"
+          },
+          "mantanat": {
+            "aggr": "aggr0"
+          }
         }
       }
+
+The following is an example of the `mac_mappings` structure that only supports
+tagging MAC addresses. Only one of `mac_mappings` or `nic_mappings` can be used
+per configuration file.
+
+    "mac_mappings": {
+      "aac3c402-3047-11e3-b451-002590c57864": {
+        "manta": "90:e2:ba:4b:ec:d1"
+      },
+      "445aab6c-3048-11e3-9816-002590c3f3bc": {
+        "manta": "90:e2:ba:4a:32:71",
+        "mantanat": "90:e2:ba:4a:32:71"
+      }
     }
 
 In a multi-datacenter configuration, this would be used to configure the
-- 
2.21.0

